core/src/main/java/com/datastax/oss/driver/api/core/CqlSession.java

```diff
@@ -77,6 +78,46 @@ default ResultSet execute(@NonNull String query) {
   /**
    * Prepares a CQL statement synchronously (the calling thread blocks until the statement is
    * prepared).
+   *
+   * <p>Note that the bound statements created from the resulting prepared statement will inherit
+   * some of the attributes of the provided simple statement. That is, given:
+   *
+   * <pre>{@code
+   * SimpleStatement simpleStatement = SimpleStatement.newInstance("...");
+   * PreparedStatement preparedStatement = session.prepare(simpleStatement);
+   * BoundStatement boundStatement = preparedStatement.bind();
+   * }</pre>
+   *
+   * Then:
+   *
+   * <ul>
+   *   <li>the following methods return the same value as their counterpart on {@code
+   *       simpleStatement}:
+   *       <ul>
+   *         <li>{@link Request#getConfigProfileName() boundStatement.getConfigProfileName()}
+   *         <li>{@link Request#getConfigProfile() boundStatement.getConfigProfile()}
+   *         <li>{@link Statement#getPagingState() boundStatement.getPagingState()}
+   *         <li>{@link Request#getRoutingKey() boundStatement.getRoutingKey()}
+   *         <li>{@link Request#getRoutingToken() boundStatement.getRoutingToken()}
+   *         <li>{@link Request#getCustomPayload() boundStatement.getCustomPayload()}
+   *         <li>{@link Request#isIdempotent() boundStatement.isIdempotent()}
+   *         <li>{@link Request#getTimeout() boundStatement.getTimeout()}
+   *         <li>{@link Statement#getPagingState() boundStatement.getPagingState()}
+   *         <li>{@link Statement#getPageSize()} boundStatement.getPageSize()}
```

Comment:
Nit: for these next 4, there is an extraneous `}` after `()` which i think would cause formatting issues.